components/expand-collapse/expand-collapse.js

```diff
@@ -0,0 +1,144 @@
+				display: none;
+				overflow: hidden;
+				transition: height 400ms cubic-bezier(0, 0.7, 0.5, 1);
+			}
+
+			.d2l-expand-collapse-container:not([data-state="collapsed"]) {
+				display: block;
+			}
+
+			.d2l-expand-collapse-container[data-state="expanded"] {
+				overflow: visible;
+			}
+
+			/* prevent margin colapse on slotted children */
+			.d2l-expand-collapse-content:before,
+			.d2l-expand-collapse-content:after {
+				content: ' ';
+				display: table;
+			}
+
+			@media (prefers-reduced-motion: reduce) {
+				.d2l-expand-collapse-container {
+					transition: none;
+				}
+			}
+		`;
+	}
+
+	constructor() {
+		super();
+		this.expanded = false;
+		this._state = states.COLLAPSED;
+		this._height = '0';
+	}
+
+	updated(changedProperties) {
+		if (changedProperties.has('expanded')) {
+			this._expandedChanged(this.expanded);
+		}
+	}
+
+	render() {
+		const styles = { height: this._height };
+		return html`
+			<div class="d2l-expand-collapse-container" data-state=${this._state} @transitionend=${this._onTransitionEnd} style=${styleMap(styles)}>
+				<div class="d2l-expand-collapse-content">
+					<slot></slot>
+				</div>
+			</div>
+		`;
+	}
+
+	async _expandedChanged(val) {
+		if (val) {
+			if (reduceMotion) {
+				this._state = states.EXPANDED;
+				this._height = 'auto';
+			} else {
+				this._state = states.PREEXPANDING;
+				await this.updateComplete;
+				await new Promise((r) => requestAnimationFrame(() => requestAnimationFrame(r)));
+				if (this._state === states.PREEXPANDING) {
+					this._state = states.EXPANDING;
+					const content = this._getContent();
+					this._height = `${content.scrollHeight}px`;
+				}
+			}
+		} else {
+			if (reduceMotion) {
+				this._state = states.COLLAPSED;
+				this._height = '0';
+			} else {
+				this._state = states.PRECOLLAPSING;
+				const content = this._getContent();
+				this._height = `${content.scrollHeight}px`;
+				await this.updateComplete;
+				await new Promise((r) => requestAnimationFrame(() => requestAnimationFrame(r)));
+				if (this._state === states.PRECOLLAPSING) {
+					this._state = states.COLLAPSING;
+					this._height = '0';
+				}
+			}
+		}
+	}
+
+	_getContent() {
+		return this.shadowRoot.querySelector('.d2l-expand-collapse-content');
+	}
+
+	async _getUpdateComplete() {
+		const fontsPromise = document.fonts ? document.fonts.ready : Promise.resolve();
+		await super._getUpdateComplete();
+		/* wait for the fonts to load because browsers have a font block period
+		where they will render an invisible fallback font face that may result in
+		improper height calculations if the expand-collapse starts expanded */
+		await fontsPromise;
+	}
+
+	_onTransitionEnd() {
+		if (this._state === states.EXPANDING) {
+			this._state = states.EXPANDED;
+			this._height = 'auto';
+		} else if (this._state === states.COLLAPSING) {
+			this._state = states.COLLAPSED;
+		}
+	}
+
+}
+customElements.define('d2l-expand-collapse', ExpandCollapse);
```

Comment:
In terms of the component name... wondering what folks think. This component will end up being paired with an opener and likely wrapped in a visible "box" of some kind, so maybe its name should be a little more narrow in scope? Example: `d2l-expand-collapse-content`

Reply:
I like that, it aligns it a bit more closely with dropdown.

Reply:
I can also see a future where some of the wrapper components -- like the one FACE is using -- also get pulled into core and could live in this directory. So the directory could still be `expand-collapse` but then we'd have a few bits and bobs in here for people to use.